src/scanoss/inspection/license_summary.py

```
@@ -0,0 +1,163 @@
+    """
+       SCANOSS LicenseSummary class
+       Inspects results and generates comprehensive license summaries from detected components.
+
+       This class processes component scan results to extract, validate, and aggregate license
+       information, providing detailed summaries including copyleft analysis and license statistics.
+       """
+
+    # Define required license fields as class constants
+    REQUIRED_LICENSE_FIELDS = ['spdxid', 'url', 'copyleft', 'source']
+
+    def __init__( # noqa: PLR0913
```

Pylint (3.3.7)

[refactor] Too many arguments (10/5)

(R0913)